code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -22,22 +22,30 @@
 		gulp_size = max(round(reagents.total_volume / 5), 5)
 
 /obj/item/reagent_containers/food/drinks/attack(mob/living/M, mob/user, def_zone)
-
 	if(!reagents || !reagents.total_volume)
-		to_chat(user, "<span class='warning'>[src] is empty!</span>")
-		return 0
+		if(iscyborg(user))
+			var/mob/living/silicon/robot/R = user
+			R.synthetize_reagents() //maybe this will work?
```

Comment:
This proc actually belongs to the module rather than the mob, so you'll need to get the module to call it.

Reply:
The intention here is to make it trigger when you use a drinkingglass (which is what the service borg starts with). If the drinking glass is empty and you try to drink from it (which silicons can't do, I guess), it will instead trigger the synthetize proc.

This way if I add the same lines to the beaker, I can make it work with the Medical bot (I think it will make the medical bot a nice support for the chemist)

Reply:
Or am I not understanding what you are suggesting?

Reply:
Cyborgs do not have the proc synthetize_reagents(). You'll need to have the cyborg's module instead of the cyborg call the proc, but you should make the other module change I mentioned first.

Reply:
If you add the proc to `/obj/item/robot_module/proc/`, you will need to create a `/obj/item/robot_module/` variable to call the proc. It does not work here because a `/mob/living/silicon/robot` is calling the proc which it does not have.

Reply:
This makes so much sense that I feel ashamed for my code. Will fix this

Reply:
Fixed it